### ISSUE TITLE
Memory alignment for older c++ versions too

### ISSUE THREAD

**Issue by olesenm** [ROCm/OpenFOAM_HMM#1]
Hi Leopold,

For List.C, would prefer something that also works with older versions.
Eg,

```
#include <new>

    if (len > 0)
    {
        // With sign-check to avoid spurious -Walloc-size-larger-than

        T* nv;

        #if (__cplusplus >= 201703L)
        if (sizeof(T)*len > 200)
        {
            nv = new (std::align_val_t(256)) T[len];
        }
        else
        {
            nv = new (std::align_val_t(16)) T[len];
        }
        #else
        nv = new T[len];
        #endif

...
```

**Comment by leopold-grinberg** [comment]
LG submitted a merge request .

**Comment by olesenm** [comment]
@leopold-grinberg 
In the first check in, there are numerous places where `min(..)` has been qualified as `Foam::min(..)` - presumably to avoid ambiguity with std::min.  Why is this needed? Unless someone has a 'using namespace std' that is swamping everything.  If that is the case, perhaps not best practice.

Qualifying Foam::min() is really not particularly convenient.

**Comment by leopold-grinberg** [comment]
@olesenm   you are correct ... we did observe a collision with std namespace and compiler was basically giving up 
on compiling the code .... so i had to add Foam::min   Foam::max  to go around this issue .
I am open for any alternative design ...  and for helping to investigate where the "using namespace std"
is coming from .  It was not really obvious to us where it came from ...

**Comment by ppanchad-amd** [comment]
Hi @olesenm. Has your issue been resolved? If so, please close the ticket. Thanks!

**Comment by olesenm** [comment]
Sure, lets close it out and reopen elsewhere as needed.
